pkg/operator/sync_controllermanager_v311_00.go

```diff
@@ -216,3 +332,33 @@ func manageServiceCatalogControllerManagerDeployment_v311_00_to_latest(client ap
 
 	return resourceapply.ApplyDaemonSet(client, recorder, required, resourcemerge.ExpectedDaemonSetGeneration(required, generationStatus), forceRollout)
 }
+
+func addProxyToEnvironment(required *appsv1.DaemonSet, proxyConfig *configv1.Proxy) {
+	required.Spec.Template.Spec.Containers[0].Env = append(required.Spec.Template.Spec.Containers[0].Env,
+		[]corev1.EnvVar{
+			{
+				Name:  httpProxyEnvVar,
+				Value: proxyConfig.Status.HTTPProxy,
+			},
+			{
+				Name:  httpsProxyEnvVar,
+				Value: proxyConfig.Status.HTTPSProxy,
+			},
+			{
+				Name:  noProxyEnvVar,
+				Value: proxyConfig.Status.NoProxy,
+			},
+			{
+				Name:  strings.ToLower(httpProxyEnvVar),
+				Value: proxyConfig.Status.HTTPProxy,
+			},
+			{
+				Name:  strings.ToLower(httpsProxyEnvVar),
+				Value: proxyConfig.Status.HTTPSProxy,
+			},
+			{
+				Name:  strings.ToLower(noProxyEnvVar),
+				Value: proxyConfig.Status.NoProxy,
+			},
```

Comment:
`os.Getenv("HTTP_PROXY")` returns `""` for both unset `HTTP_PROXY` and for `HTTP_PROXY=""`. Same thing for `proxyConfig.Status.NoProxy` it will be empty string since that's the default for the string.

I think it will be okay to have them there. I know `openshift-ansible` sets them to "" during install because our broker had a problem with them being there :) we had to actually check to make sure the proxy had a non-zero length before using it.

Reply:
I was thinking about the possibility of the daemonset doing something like `v, isSet := os.LookupEnv("HTTP_PROXY")`, and then doing something based on `isSet`.

It seems unlikely that there would be any significant difference between unset and `""`, but just wanted to make sure :slightly_smiling_face:

Reply:
For now, I'd like to leave it as is. And address in another PR if there is a problem found in testing, if that's okay.